### PR TITLE
Creating BOM (google-cloud-bom)

### LINKED ISSUE #2307: Create BOM for google-cloud-java
Example BOM:

https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-dependencies/pom.xml

Here is how Spring Cloud uses the BOM for their dependencies:

https://spring.io/blog/2016/09/26/spring-cloud-camden-release-and-brixton-sr6-are-available

The work for us is the following:

1. Extract dependencyManagement and pluginManagement to a dedicated pom.xml (e.g., artifact: google-cloud-java-bom)
2. In the parent pom.xml, import the BOM

Then consumers can import the BOM too, instead of specifying individual versions.

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Fixes #2307

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
cc @saturnism

**Comment by garrettjonesgoogle** [comment]
Fixes #2307

File: pom.xml [hunk 1]
```diff
@@ -136,804 +136,21 @@
...
-        <artifactId>truth</artifactId>
-        <version>0.30</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.api</groupId>
-        <artifactId>gax-grpc</artifactId>
-        <version>${gax-grpc.version}</version>
-	<classifier>testlib</classifier>
+        <artifactId>google-cloud-dependencies</artifactId>
+        <version>${google-cloud-parent.version}</version>
```
Comment (vam-google) [inline pom.xml]: This is confusing: the artifact is called `google-cloud-dependencies`, but the version is `google-cloud-parent.version`

File: google-cloud-dependencies/pom.xml [hunk 2]
```diff
@@ -0,0 +1,810 @@
+<?xml version="1.0"?>
```
Comment (vam-google) [inline google-cloud-dependencies/pom.xml]: Should be this module explicitly called `google-cloud-bom`?

Comment (garrettjonesgoogle) [reply google-cloud-dependencies/pom.xml]: Yeah I think that is clearer.

File: pom.xml (hunk 1 above)
Comment (garrettjonesgoogle) [reply pom.xml]: Will fix.

**Comment by garrettjonesgoogle** [comment]
I have addressed feedback and verified that update_versions.sh works. PTAL

**Comment by garrettjonesgoogle** [comment]
I agree with the comment you made in https://github.com/GoogleCloudPlatform/google-cloud-java/pull/2332 , that it's good to have compile and test dependencies versioned together. I think we can consider pulling out test deps as a later cleanup. Anyway, Spring (our spiritual leader in this matter) has both compile and test deps together: https://github.com/spring-io/platform/blob/master/platform-bom/pom.xml .

**Comment by garrettjonesgoogle** [comment]
I have moved pure test dependencies back to the main pom.xml, PTAL

**Comment by joaoandremartins** [comment]
This needs documentation that shows how to use the BOM (with Maven at least, also possible with Gradle with a special plugin), which version of the BOM to use, etc.

**Comment by garrettjonesgoogle** [comment]
@jabubake already requested a readme in https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2708 (let's continue any discussion there)

**Pull request merged**